bin/run.sh

```diff
@@ -31,20 +31,40 @@ mkdir -p "${output_dir}"
 
 echo "${slug}: testing..."
 
+
+TESTFILE=$(jq -r .files.test[0] ${input_dir}/.meta/config.json)
+TEST="${input_dir}/${TESTFILE}"
+
 # Run the tests for the provided implementation file and redirect stdout and
 # stderr to capture it
-# TODO: Replace 'RUN_TESTS_COMMAND' with the command to run the tests
-test_output=$(RUN_TESTS_COMMAND 2>&1)
+ln -sf $(realpath ./vendor) ${input_dir}/vendor
+rm -f $results_file
+# rewrite any skipped tests to not be skipped
+sed -i'' -e 's/skip.test/do.test/' $TEST
+test_output=$(wren_cli $TEST $results_file 2>&1)
+
+status=$?
+
+# cleanup
+rm ${input_dir}/vendor
+# echo "$test_output"
+
+# if we have a results file that was generated by the test runner
+# then use that, augment with stack trace if necessary
+if [ -f $results_file ]; then
+    jq -r '.' $results_file | sponge $results_file
 
-# Write the results.json file based on the exit code of the command that was 
-# just executed that tested the implementation file
-if [ $? -eq 0 ]; then
-    jq -n '{version: 1, status: "pass"}' > ${results_file}
+    index=$(cat $results_file | jq -M '[.tests[].status] | index("fail")')
+    if [ "$index" != "null" ]; then
+        trace=$(echo "$test_output" | sed -e '1,/STACKTRACE/ d' | sed "s%${input_dir}%.%")
+        jq ".tests[$index].message |= \"$trace\"" $results_file | sponge $results_file
+    fi
+# otherwise something very bad happened so we return status:error
 else
     # OPTIONAL: Sanitize the output
     # In some cases, the test output might be overly verbose, in which case stripping
     # the unneeded information can be very helpful to the student
-    # sanitized_test_output=$(printf "${test_output}" | sed -n '/Test results:/,$p')
+    sanitized_test_output=$(echo "${test_output}" | sed "s%${input_dir}%.%" )
```

Comment:
Is there some magic to printf vs echo here? @ErikSchierboom

Reply:
I forget why I changed it.

Reply:
I can't even remember :D